src/modules/DNS.rb

```diff
@@ -381,65 +330,40 @@ def Write
         _("Update /etc/resolv.conf")
       ]
 
+      # ensure that nothing is saved in case old values are the same, as it makes
+      # rcnetwork reload restart all interfaces (even 'touch /etc/sysconfig/network/dhcp'
+      # is sufficient)
+      update_sysconfig_dhcp
+
+      log.info("DNS: Writing configuration")
+      if !@modified
+        log.info("No changes to DNS -> nothing to write")
+        return true
+      end
+
       # Write dialog caption
       caption = _("Saving Hostname and DNS Configuration")
-      sl = 0 # 100; for testing
-
-      Progress.New(caption, " ", Builtins.size(steps), steps, [], "")
 
-      # Allow to set hostname even if it's modified by DHCP (#13427)
-      # if(NetworkConfig::DHCP["DHCLIENT_SET_HOSTNAME"]:false != true) {
+      Progress.New(caption, " ", Builtins.size(steps), steps, [], "") if gui
 
       # Progress step 1/3
-      ProgressNextStage(_("Writing hostname..."))
-
-      # change the hostname
-      SCR.Execute(path(".target.bash"), Ops.add("/bin/hostname ", @hostname))
-
-      # write hostname
-      SCR.Write(
-        path(".target.string"),
-        HOSTNAME_PATH,
-        Ops.add(fqhostname, "\n")
-      )
-
-      create_hostname_link
-
-      Builtins.sleep(sl)
+      ProgressNextStage(_("Writing hostname...")) if gui
+      update_hostname
+      Builtins.sleep(0) if gui
```

Comment:
hm, consequence of some massive updating and code shifting and so on. I'll drop it.